server/testnet.go

```diff
@@ -0,0 +1,240 @@
+/*
+ * Ported from Tendermint
+ */
+package server
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	cfg "github.com/tendermint/tendermint/config"
+	gaiacfg "github.com/cosmos/cosmos-sdk/config"
+	"github.com/tendermint/tendermint/p2p"
+	"github.com/tendermint/tendermint/types"
+	pvm "github.com/tendermint/tendermint/types/priv_validator"
+	cmn "github.com/tendermint/tmlibs/common"
+	"github.com/cosmos/cosmos-sdk/wire"
+)
+
+var (
+	nValidators    int
+	nNonValidators int
+	outputDir      string
+	nodeDirPrefix  string
+
+	populatePersistentPeers bool
+	hostnamePrefix          string
+	startingIPAddress       string
+	p2pPort                 int
+)
+
+const (
+	nodeDirPerm = 0755
+)
+
+func init() {
+}
+
+// get cmd to initialize all files for tendermint testnet and application
+func TestnetFilesCmd(ctx *Context, cdc *wire.Codec, appInit AppInit) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "testnet",
+		Short: "Initialize files for a Gaiad testnet",
+		Long: `testnet will create "v" + "n" number of directories and populate each with
+necessary files (private validator, genesis, config, etc.).
+
+Note, strict routability for addresses is turned off in the config file.
+
+Optionally, it will fill in persistent_peers list in config file using either hostnames or IPs.
+
+Example:
+
+	gaiad testnet --v 4 --o ./output --populate-persistent-peers --starting-ip-address 192.168.10.2
+	`,
+		RunE: func(_ *cobra.Command, _ []string) error {
```

Comment:
RunE should not define the function in this situation - it's just too large and it get's confusing with all the added indentation. Define a separate function and take it as an argument. Ex:

```golang 
type inp struct {
     cdc *wire.Codec, 
     appInit AppInit,
}

func TestnetFilesCmd(ctx *Context, cdc *wire.Codec, appInit AppInit) *cobra.Command {
     inp := inp{cdc, appInit}

     cmd := &cobra.Command{
		Use:     "testnet",
		Short:  "Initialize files for a Gaiad testnet",
		RunE:   inputs.gregsFunc, 
}

func (c inp)gregsFunc(_ *cobra.Command, _ []string) error {
     something that uses c.cdc
     something else that uses c.appInit
}
```